src/pages/Application/View/[id].tsx

```diff
@@ -9,83 +9,77 @@ import withContext from '../../../shared/HOC/withContext';
-    };
-  }
+  // The type of user viewing this page (Organzier, Sponsor_TIER, etc.)
+  const [userType, setUserType] = useState<UserType>(UserType.UNKNOWN);
 
-  public async componentDidMount() {
-    try {
-      const viewer = (await Account.getSelf()).data.data;
-      console.log(viewer, viewer.accountType);
-      this.setState({ userType: viewer.accountType });
-      // tslint:disable-next-line:no-empty
-    } catch (e) { }
-    try {
-      const hacker = (await Hacker.get(this.props.match.params.id)).data.data;
-      const account = (await Account.get(hacker.accountId as string)).data.data;
-      hacker.accountId = account;
-      this.setState({ hacker, isLoading: false });
-    } catch (e) {
-      this.setState({ isLoading: false });
-    }
-  }
+  // Get the id param from url's query
+  let { id } = useParams();
+
+  const nomineeContext = useContext(NomineeContext);
 
-  public render() {
-    if (this.state.hacker) {
-      return (
-        <Flex justify-content={'center'} m={'10px'} flexDirection={'column'}>
-          <Box m={'auto'}>
-            <SingleHackerView
-              hacker={this.state.hacker}
-              userType={this.state.userType}
-            />
-            {this.context && this.context.nominees && <hr />}
-            {this.context && this.context.nominees && (
+  // When this component mounts, load hackers data
+  useEffect(() => {
+    (async () => {
+      try {
+        const viewer = (await Account.getSelf()).data.data;
+        console.log(viewer, viewer.accountType);
+        setUserType(viewer.accountType);
+        const hacker = (await Hacker.get(id)).data.data;
+        const account = (await Account.get(hacker.accountId as string)).data.data;
+        hacker.accountId = account;
+        setHacker(hacker);
+      } finally {
+        setIsLoading(false);
+      }
+    })();
+  }, []);
+
+  if (hacker) {
+    return (
+      <Flex justify-content={'center'} m={'10px'} flexDirection={'column'}>
+        <Box m={'auto'}>
+          <SingleHackerView
+            hacker={hacker}
+            userType={userType}
+          />
+          {nomineeContext && nomineeContext.nominees && (
+            <>
+              <hr />
               <Flex m={'auto'}>
                 <Box>
                   <H2 marginBottom={'3px'}>Save Hacker:</H2>
                 </Box>
                 <Box>
-                  <HackerSelect hackerId={this.state.hacker.id} />
+                  <HackerSelect hackerId={hacker.id} />
                 </Box>
               </Flex>
-            )}
-          </Box>
-        </Flex>
-      );
-    } else if (this.state.isLoading) {
-      return (
-        <Flex alignItems={'center'}>
-          <Box m={'auto'}>
-            <H1>Loading...</H1>
-          </Box>
-        </Flex>
-      );
-    } else {
-      return <Redirect to={'/404'} />;
-    }
+            </>
+          )}
+        </Box>
+      </Flex>
+    );
+  }
+
+  if (isLoading) {
```

Comment:
what's the reason to make `isLoading` its own `if` statement than its previous `else if`?

Reply:
`if`, `else if`, `else` does sound structurally better than separate if blocks, even if they are functionally equivalent because of the return. I'll update the HackPass page to match this as well.

Reply:
based on what we talked about, can you create a new issue and comment here for reference?